src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesLegendURLTest.java

```diff
@@ -404,8 +404,37 @@ public void testOnlineResourceWidthHeight() throws Exception {
         Element onlineResource = (Element) onlineResources.item(0);
         String href = onlineResource.getAttribute("xlink:href");
         assertNotNull(href);
-        assertTrue(href.contains("width=20"));
-        assertTrue(href.contains("height=20"));
+        assertTrue(href.contains("width=50"));
```

Comment:
So basic polygons has a custom legend? Maybe add comments to clarify.

Reply:
If not a custom legend, then maybe we have a problem... I think the width and height in the request are relative to the size of the icons, not the size of the returned legend. They end up being the same only if a custom legend has been applied.

Reply:
fixed

Reply:
I don't see the fix. Did you try interactively layers that do not have a custom legend, and verified the output is still the same? I don't think it will.

Reply:
sorry the mess @aaime, now I got your point. Everything should be fine now.